app/models/order_cycle.rb

```diff
@@ -240,7 +240,8 @@ def exchanges_carrying(variant, distributor)
   end
 
   def exchanges_supplying(order)
-    exchanges.supplying_to(order.distributor).with_any_variant(order.variants.map(&:id))
+    variants_relation = Spree::Variant.joins(:line_items).merge(Spree::LineItem.in_orders(order))
```

Comment:
with_any_variant takes variant ids
why not simply:
variants_relation = Spree::LineItem.in_orders(order).select(:variant_id)